realm/realm-library/src/main/java/io/realm/RealmInteger.java

```diff
@@ -16,31 +16,32 @@
 package io.realm;
 
 import io.realm.annotations.Beta;
+import io.realm.internal.ManagableObject;
+import io.realm.internal.counters.UnmanagedRealmInteger;
 
 
 /**
  * TODO: document the behaviour of this object, as soon as the implementation defines it.
  */
 @Beta
-public final class RealmInteger extends Number implements Comparable<RealmInteger> {
-    private long value;
+public abstract class RealmInteger extends Number implements Comparable<RealmInteger>, ManagableObject {
 
     /**
      * Creates a new {@code RealmInteger}, with the specified initial value.
      *
      * @param value initial value.
      */
-    public RealmInteger(long value) {
-        this.value = value;
+    public static RealmInteger valueOf(long value) {
```

Comment:
Since this can only return a unmanaged `RealmInteger`, I don't think it is a useful API?